streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -479,10 +460,12 @@ public Subscription subscription(final Set<String> topics) {
 
             for (final InternalTopologyBuilder.TopicsInfo topicsInfo : topicGroups.values()) {
                 for (final String topicName : topicsInfo.repartitionSourceTopics.keySet()) {
-                    int numPartitions = repartitionTopicMetadata.get(topicName).numPartitions;
+                    int numPartitions = UNKNOWN;
+                    try {
+                        numPartitions = repartitionTopicMetadata.get(topicName).numberOfPartitions();
```

Comment:
I think we can just relax

```
public int numberOfPartitions() {
        if (numberOfPartitions == -1) {
            throw new IllegalStateException("Number of partitions not specified.");
        }
        return numberOfPartitions;
    }
```

To allow it returning UNKNOWN.

Reply:
Ack